BSPConversionLib/Source/EntityConverter.cs

```diff
@@ -137,6 +143,88 @@ private HashSet<string> GetGiveTargets()
 			return targets;
 		}
 
+		private void ConvertFuncDoor(Entity entity)
+		{
+			GetMoveDir(entity);
+
+			if (float.TryParse(entity["health"], out var health))
+			{
+				entity.ClassName = "func_button"; // Health is obsolete on func_door, maybe fix in engine and update this
+			}
+		}
+
+		private void ConvertFuncButton(Entity entity)
+		{
+			GetMoveDir(entity);
+
+			GetButtonFlags(entity);
+
+			if (entity["wait"] == "-1")
+			{
+				entity["wait"] = "0.001";
+			}
+
+			var targets = GetTargetEntities(entity);
+
+			foreach (var target in targets)
+			{
+				switch (target.ClassName)
+				{
+					case "func_door":
+						ButtonDoorLink(entity, target);
+						break;
+				}
+			}
+		}
+
+		private static void ButtonDoorLink(Entity entity, Entity target)
+		{
+			var connection = new Entity.EntityConnection()
+			{
+				name = "OnPressed",
+				target = target["targetname"],
+				action = "Open",
+				param = null,
+				delay = 0,
+				fireOnce = -1
+			};
+			entity.connections.Add(connection);
+		}
+
+		private static void GetButtonFlags(Entity entity)
+		{
+			float.TryParse(entity["speed"], out var speed);
+
+			if ((speed == -1 || speed >= 9999) && (entity["wait"] == "-1")) // TODO: Add customization setting for the upper bounds potentially?
+			{
+				entity["spawnflags"] = "1"; // Don't move flag
+			}
+			if (!float.TryParse(entity["health"], out var health) || entity["health"] == "0")
+			{
+				entity["spawnflags"] = "256"; // Press on touch
+			}
+		}
+
+		private static void GetMoveDir(Entity entity)
```

Comment:
This should be renamed to `SetMoveDir`. You can leave `entity` alone since it can either be a button or door.